### PR TITLE
Update wxPython to 4.2.2, using 32-bit wheel hosted on PyPI

### LINKED ISSUE #17175: Update wxPython to 4.2.2, using 32-bit wheel hosted on PyPI
Part of component updates for NVDA 2025.1 development milestone:

### Is your feature request related to a problem? Please describe.
NVDA 2024.x uses an alpha version of wxPython 4.2.x hosted on NV Access dependencies repository. While this did provide a ready to use 32-bit wheel, using a development snapshot of wxPython was risky and the project could not take advantage of changes intorduced in later builds.

### Describe the solution you'd like
wxPython 4.2.2 was released on September 11, 2024. The biggest change is restoration of 32-bit wheel so NV Access no longer needs to host its own 32-bit wheel for this dependency. This also allows the project to use newer changes from wxPython (wxWidgets 3.2.6).

### Describe alternatives you've considered
Keep using current wxPython 4.2.x alpha build.

### Additional context
Part of major component updates in line with year.1 release - wxPython is the GUI toolkit used in NVDA, so this dependency is a critical component for the screen reader. Besides, wxPython 4.2.2 introduces support for Python 3.13 (a 32-bit wheel was also built for 3.13).

Thanks.

### PR DESCRIPTION
**Pull request by dpy013**
### Link to issue number:
fix #17175 
### Summary of the issue:
Update wxPython to 4.2.2, using 32-bit wheel hosted on PyPI
### Description of user facing changes
Not yet.
### Description of development approach
Replace the wx python version of the requirements file with pip's
and tested it with a locally compiled build!
### Testing strategy:
Need to watch appveyor build after compilation
### Known issues with pull request:

Nothing yet.

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [ ] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [ ] API is compatible with existing add-ons.
- [ ] Security precautions taken.

@coderabbitai summary
Here are the test logs for the NVDA compilation builds
[nvda.log](https://github.com/user-attachments/files/17009988/nvda.log)

### CODE REVIEW

**Comment by dpy013** [comment]
Tested wx-python 4.2.2 with pr17181-34014, who has the same messages as in the test log above.
No error messages were reported.
[nvda.log](https://github.com/user-attachments/files/17011121/nvda.log)

**Comment by CyrilleB79** [comment]
I'd say that tests should be more thourough:
* Try to open and validate many dialogs off different types
* Have a visual confirmation that the layout of the main dialogs (if not all) has not been altered.

**Comment by ABuffEr** [comment]
Hi,
check also the symbol/punctuation dialog and emoji representation, that is the main reason we have used that alpha of wxPython, see #15714.

**Comment by seanbudd** [comment]
Can you please make sure to update `nvda-misc-deps` to remove the wxPython wheel

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ Add-ons will need to be re-tested and have their manifest updated.
 * Component updates:
   * Updated Ruff to 0.6.3. (#17102)
   * Updated Comtypes to 1.4.6. (#17061, @LeonarddeR)
+  * Update wxPython to 4.2.2. (#17181, @dpy013)
```
Comment (SaschaCowley) [inline user_docs/en/changes.md:46]:
```suggestion
  * Updated wxPython to 4.2.2. (#17181, @dpy013)
```

**Comment by dpy013** [comment]
Hello everyone,
I wish you all a happy Mid-Autumn Festival. I have completed all the changes mentioned above.
https://github.com/nvaccess/nvda-misc-deps/pull/34

**Review by SaschaCowley — APPROVED** [review]

**Pull request merged**